ouroboros-consensus-test/test-storage/Test/Ouroboros/Storage/LedgerDB/OnDisk.hs

```diff
@@ -348,10 +685,10 @@ runMock cmd initMock =
     cfg = extLedgerDbConfig (mockSecParam initMock)
 
     go :: Cmd MockSnap -> Mock -> (Success MockSnap, Mock)
-    go Current       mock = (Ledger (cur (mockLedger mock)), mock)
+    go Current       mock = (Ledger (stowLedgerTables $ cur (mockLedger mock)), mock)
```

Comment:
Making a note for myself here. The problem that we have is that:

```haskell
Ledger (ExtLedgerState TestBlock EmptyMK)
```

Here we could simply drop the tables. However, the problem is that we wouldn't be comparing the ledger tables in both states (which at the moment are part of `PayloadDependentState Tx mk` (and therefore of the ledger state). This might not be a big problem since deviations in the ledger tables might be eventually caught up, but I don't have a solid argument at the moment on why this will be the case.

If we change the type of the `Ledger` constructor above to use `ValuesMK` the problem is that we have at the moment no way of getting the current ledger state of the ledger DB which includes the `ValuesMK`. So we won't be able to easily implement:

```haskell
go _ Current =
        atomically $ (Ledger . ledgerDbCurrent . snd) <$> readTVar dbState
```

So dropping the tables on the mock seems to be the only alternative I see that won't require a large change in these tests and the `LedgerDB` API.

Reply:
Add follow up action: add a command that returns all the ledger tables. We can implement this in the SUT by using a handle to the backing store we create in these tests.

Reply:
See also https://github.com/input-output-hk/ouroboros-network/blob/5e03b0087d9607c108d9ca59c407a7c25d8284d4/ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/Ledger/Query.hs#L298-L309